src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -1002,6 +1026,211 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_JoinerStart(voi
     return error;
 }
 
+#if CHIP_DEVICE_CONFIG_ENABLE_THREAD_SRP_CLIENT
+
+static_assert(OPENTHREAD_API_VERSION >= 80, "SRP Client requires a more recent OpenThread version");
+
+template <class ImplClass>
+void GenericThreadStackManagerImpl_OpenThread<ImplClass>::OnSrpClientNotification(otError aError,
+                                                                                  const otSrpClientHostInfo * aHostInfo,
+                                                                                  const otSrpClientService * aServices,
+                                                                                  const otSrpClientService * aRemovedServices,
+                                                                                  void * aContext)
+{
+    switch (aError)
+    {
+    case OT_ERROR_NONE: {
+        ChipLogProgress(DeviceLayer, "OnSrpClientNotification: Last requested operation completed successfully");
+
+        if (aRemovedServices)
+        {
+            otSrpClientService * service = const_cast<otSrpClientService *>(aRemovedServices);
+            otSrpClientService * next    = nullptr;
+
+            // Free memory for all removed services.
+            do
+            {
+                next = service->mNext;
+                // Clear also names associated with the SrpClient::Service object
+                service->mInstanceName[0] = '\0';
```

Comment:
The code doesn't do what you probably intended it to do. It only clears the pointer which you anyway clear in the following `memset`.

Reply:
I'm not sure I understand. `service` is `otSrpClientService` type and it has a field mInstanceName, which is a pointer set to point at `Service mInstanceName` array. So when we set `service->mInstanceName[0] = '\0\'`, it actually sets the first element of `Service mInstanceName` array. It is not the same as setting `service->mInstanceName` to nullptr, as it will just remove address from pointer and will not clear `Service mInstanceName` array. Then when calling `if (strcmp(service.mInstanceName, "") == 0)` we got the first element equal to `'\0'` and condition is true.

Reply:
Right, I must be tired :) It's however weird that the compiler allows you to modify a constant. Or in newer OT it's no longer `const char*`? https://openthread.io/reference/struct/ot-srp-client-service?authuser=1#minstancename

Reply:
It is constant, the thing is that we do not modify it :) Pointer is just an address of object, so we do not change address, but only object under that address. So from the pointer point of view, we do not modify it.

Reply:
But `const char*` is a pointer to constant. The code:
```
void f(const char* a)
{
    a[0] = '\0';
}
```
won't build. So why your code does? :)

Reply:
Oh now I see, I guess we both were looking on arguments, but here I did const_cast https://github.com/kkasperczyk-no/connectedhomeip/pull/21/files#diff-837473660e25fab2523b543759412eb6ccd77f7c790b41126811387fa7916b20R1047